rules/lowercase_description.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const ruleMsg = 'it() description should begin with lowercase';
+
+const isItOrTestorDescribeFunction = node => {
+  return (
+    node.type === 'CallExpression' &&
+    node.callee &&
+    (node.callee.name === 'it' ||
+      node.callee.name === 'test' ||
+      node.callee.name === 'describe')
+  );
+};
+
+const isItDescription = node => {
+  const foo =
+    node.arguments &&
+    node.arguments[0] &&
+    (node.arguments[0].type === 'Literal' ||
+      node.arguments[0].type === 'TemplateLiteral');
+  return foo;
+};
+
+const testDescription = node => {
+  const { type } = node.arguments[0];
+  if (type === 'Literal') {
+    return node.arguments[0].value;
+  }
+  if (type === 'TemplateLiteral') {
+    return node.arguments[0].quasis[0].value.raw;
+  }
+};
+
+const descriptionBeginsWithLowerCase = node => {
+  if (isItOrTestorDescribeFunction(node) && isItDescription(node)) {
+    const description = testDescription(node);
+    return description[0] === description[0].toLowerCase();
```

Comment:
Looking at the definition of `testDescription()`, it can technically return undefined if the first argument's type is not `Literal` or `TemplateLiteral`, or if the `Literal` provided is not a string.

For example, the following code:

```js
it(42, () => {})
```

will result in `TypeError: Cannot read property 'toLowerCase' of undefined` since the first argument of `it()` is a `Literal`, but it's not a string.

While passing a number as the first argument of `it()` isn't really valid, I think this rule should protect against a possibility like this.

Reply:
Good catch, updated and added some tests to check this as well.